lib/Runtime/Library/JavascriptFunction.cpp

```diff
@@ -983,15 +975,24 @@ namespace Js
         return JavascriptFunction::CallSpreadFunction(function, args, spreadIndices);
     }
 
-    uint32 JavascriptFunction::GetSpreadSize(const Arguments args, const Js::AuxArray<uint32> *spreadIndices, ScriptContext *scriptContext)
+    ArgSlot JavascriptFunction::GetSpreadSize(const Arguments args, const Js::AuxArray<uint32> *spreadIndices, ScriptContext *scriptContext)
     {
         // Work out the expanded number of arguments.
-        uint32 totalLength = args.Info.Count - spreadIndices->count;
+        if (args.Info.Count < spreadIndices->count)
+        {
+            ::Math::DefaultOverflowPolicy();
+        }
+        ArgSlot spreadArgsCount = (ArgSlot)spreadIndices->count;
+        ArgSlot totalLength = (ArgSlot)args.Info.Count - spreadArgsCount;
         ::Math::RecordOverflowPolicy overflow;
-        for (unsigned i = 0; i < spreadIndices->count; ++i)
+        for (unsigned i = 0; i < spreadArgsCount; ++i)
         {
             uint32 elementLength = JavascriptArray::GetSpreadArgLen(args[spreadIndices->elements[i]], scriptContext);
-            totalLength = UInt32Math::Add(totalLength, elementLength, overflow);
+            if (elementLength > USHORT_MAX)
+            {
+                ::Math::DefaultOverflowPolicy();
```

Comment:
DefaultOverflowPolicy will lead to failfast. shouldn't you want to throw javascript range error in this case?

Reply:
Fixed